src/ansiblelint/file_utils.py

```diff
@@ -118,12 +127,24 @@ def __init__(
             if role.exists:
                 self.role = role.name
 
-        self.kind = kind or kind_from_path(self.path)
-        # We store absolute directory in dir
-        if self.kind == "role":
-            self.dir = str(self.path.resolve())
+        if str(self.path) in ['/dev/stdin', '-']:
+            self.file = NamedTemporaryFile(mode="w+", suffix="playbook.yml")
```

Comment:
Can it be kept in-memory instead?

Reply:
Nope because I do need to feed `ansible-playbook --syntax-check` a file. The rest of the linter code could work fine with in-memory files, but not ansible itself.